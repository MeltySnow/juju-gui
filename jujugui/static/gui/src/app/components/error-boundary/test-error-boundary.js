/* Copyright (C) 2018 Canonical Ltd. */
'use strict';

const React = require('react');
const enzyme = require('enzyme');
const ErrorBoundary = require('./error-boundary');

class ComponentWithError extends React.Component {
  constructor(props) {
    super(props);
    throw new Error('Error!');
  }
  render() {
    return (
      <div>
        <h1>Buggy component</h1>
      </div>
    );
  }
}

describe('Error boundary', function() {

  const renderComponent = (options = {}) => enzyme.shallow(
    <ErrorBoundary />
  );

  it('can render', () => {
    const wrapper = renderComponent({
      content: (<span>Children content!</span>)
    });
    expect(wrapper).toMatchSnapshot();
  });

  it('should show error message if available', () => {
    const renderComponentWithError = (options = {}) => enzyme.shallow(
      <ErrorBoundary />
    );
    const wrapper = renderComponentWithError({
      content: (<ComponentWithError />)
    });
    expect(wrapper).toMatchSnapshot();
  });
});
