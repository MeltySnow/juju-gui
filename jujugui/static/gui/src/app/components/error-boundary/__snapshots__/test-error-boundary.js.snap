// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error boundary can render 1`] = `""`;

exports[`Error boundary should show error message if available 1`] = `""`;
